### DISCUSSION TITLE
How can I update daily to see the change?

### DISCUSSION THREAD

**Discussion by bluebyt** [Rafostar/clapper#15] [Q&A] [answered]
Hi, I have installed clapper with Flatpak, how can I update daily to see the change?
I tried "flatpak update" but that doesn't update to the latest clapper on github.
Command I used to install:
flatpak install https://rafostar.github.io/flatpak/com.github.rafostar.Clapper.flatpakref

**Comment by Rafostar** [comment, marked as answer]
If during install, when you were asked to add "rafostar" repo and selected "yes", than you should receive updates automatically (via `GNOME Software` store - my app should also appear there) or you can force update manually via:
```sh
flatpak update com.github.rafostar.Clapper
```

Saying all that, I do not own a server to do the `nightlies` for each git push for Flatpak version. I will be doing the recompiling and pushing updated version manually once every few new features are added and tested at least a little bit by me.

You can check on which commit latest Flatpak is currently by visiting [my flatpak repo](https://github.com/Rafostar/flatpak) and reading latest `Clapper` commit message (compare with this repo commits for a sort of a changelog).

**Comment by bluebyt** [comment]
Ok thank you very much to take the time to answer my question!

**Comment by Rafostar** [comment]
Quick update on current situation here. Clapper is now on Flathub. The 3rd party flatpak repo mentioned in above answer will no longer be maintained. Please remove your current installation and install Clapper from Flathub to get latest stable version.

**Comment by peteruithoven** [comment]
I noticed the [nightly flatpak workflows](https://github.com/Rafostar/clapper/actions/workflows/flatpak-nightly.yml) seem to work again, does this mean we kind of have a github hosted nightly channel? 

I found that after uninstalling the regular Clapper I was able to install the com.github.rafostar.Clapper-x86_64 artefact in the most recent workflow. 

Does installing / sideloading this flatpak mean I get updates when I update my flatpaks? 

I understand nightlies are unstable, but all the recent activity got me super curious and sadly since I'm on a LTS (22.04.3 LTS) I don't seem to have access to the meson version required to build it manually.

**Reply by Rafostar** [reply]
@peteruithoven 

Nightly builds are currently kinda broken performance-wise. Their main purpose is to test how things work with latest GTK and GStreamer versions. For something that is much closer to Flathub ver, please use normal (non-nightly) [git action builds](https://github.com/Rafostar/clapper/actions/workflows/flatpak.yml). Similarly to nighties, they are build on each commit, so latest unstable Clapper, but have mostly stable dependencies as in Flathub version instead of git snapshot every 24h.

> Does installing / sideloading this flatpak mean I get updates when I update my flatpaks?

No. Because you install a compiled package, but not the repo (and I do not host a repo), so they will not be automatically updated. The only way to update is to download newer build and install it on top of previous one.

On the side note, there will be a new Clapper stable release (and thus Flathub push/update) in the next few days :smile:

**Reply by peteruithoven** [reply]
Thanks for the explanations! I'll try the non nightly builds and keep an eye out for the next stable. 

Great work on the new UI, looks and works great!

**Reply by Rafostar** [reply]
Thanks. Also, if you happen to be using Intel/AMD GPU, I recommend switching to Wayland session.

**Reply by Scotty-Trees** [reply]
Just curious, will the soon to be released Clapper version have the updated about dialog windows (Adw.AboutWindow) ?

**Reply by Rafostar** [reply]
It will be using `AdwAboutWindow` but not yet `AdwAboutDialog`.
Also, for asking questions regarding Clapper, best place is its matrix channel :smile:
